prymer/variant.py

```diff
@@ -0,0 +1,259 @@
+        """
+        Convert `pysam.VariantRecord` to one or more `SimpleVariant`s.
+        """
+        # Skip symbolic ref alleles
+        if "<" in variant.ref:
+            return []
+
+        ref = variant.ref
+        alt_afs = SimpleVariant._extract_mafs(variant)
+        if alt_afs is not None and len(alt_afs) != len(variant.alts):
+            raise ValueError(f"Different number of allele frequencies vs. alt alleles: {variant}")
+
+        simple_variants: list[SimpleVariant] = []
+
+        for i, alt in enumerate(variant.alts):
+            # Skip symbolic alt alleles
+            if "<" in alt:
+                continue
+
+            if len(alt) < len(ref) and ref.startswith(alt):
+                # Deletion
+                padding_bases = len(alt)
+                deleted_bases = len(ref) - padding_bases
+                start = variant.pos + padding_bases
+                end = start + deleted_bases - 1
+                ref = ref[padding_bases:]
+                alt = ""
+            elif len(alt) > len(ref) and alt.startswith(ref):
+                # Insertion
+                padding_bases = len(ref)
+                deleted_bases = len(ref) - padding_bases
+                start = variant.pos + padding_bases - 1
+                end = start + 1
+                ref = ""
+                alt = alt[padding_bases:]
+            else:
+                # Substitution (simple or complex)
+                start = variant.pos
+                end = start + len(ref) - 1
+
+            id = variant.id or f"{variant.chrom}:{start}"
+            if len(variant.alts) > 1:
+                id = f"{id}-{i+1}/{len(variant.alts)}"
+
+            simple_variant = SimpleVariant(
+                id=id,
+                refname=variant.chrom,
+                start=start,
+                end=end,
+                ref=ref,
+                alt=alt,
+                af=alt_afs[i] if alt_afs is not None else None,
+            )
+            simple_variants.append(simple_variant)
+
+        return simple_variants
+
+    @staticmethod
+    def _extract_mafs(variant: pysam.VariantRecord) -> Optional[list[float]]:
+        """
+        Calculates/extracts alt allele frequencies (MAF) based on whether `VariantRecord` has
+        CAF, AF, AC and AN in the INFO field.
+
+        Args:
+            variant: the variant from which the MAF will be computed
+
+        Returns:
+            the list of frequencies of alternate alleles
+        """
+        if "CAF" in variant.info:
+            # Assumes CAF is a list of allele frequencies, with the first being the reference allele
+            return [float(af) for af in variant.info["CAF"][1:]]
+        elif "AF" in variant.info:
+            return [float(af) for af in variant.info["AF"]]
+        elif "AC" in variant.info and "AN" in variant.info:
+            an = float(variant.info["AN"])
+            return [float(ac) / an for ac in variant.info["AC"]]
+
```

Comment:
Why not use "FORMAT/GT" if available like previously?  I was surprised to see them gone, and didn't see a comment in the review highlighting why.

Reply:
I went through and updated it to return the actual AF per alternate allele rather than `1 - ref_allele_freq` for _each_ alt.  It felt like that would be quite a bit more work to do accurately using GTs, and I'm not sure that's ever been used.